src/tower.rs

```diff
@@ -28,6 +28,16 @@ impl Tower {
         return *self.buff_to_level.get(&BuffType::Aura).unwrap_or(&0);
     }
 
+    pub fn can_have_aura(&self, buff: &BuffType) -> bool {
+        if self.buff_to_level.keys().count() < 2 {
+            return true;
```

Comment:
`keys().count()` produces an iterator and iterates it. Better to use `len()` which is O(1).